app/merginapi.cpp

```diff
@@ -43,15 +53,26 @@ void MerginApi::listProjectsReplyFinished()
         // TODO propagate error
         QString message = QStringLiteral("Network API error: %1(): %2").arg("listProjects", r->errorString());
         qDebug("%s", message.toStdString().c_str());
+        emit networkErrorOccurred( r->errorString() );
+        QString errMsg = r->errorString() ;
+#ifdef ANDROID
+        QtAndroid::runOnAndroidThread([errMsg] {
+            QAndroidJniObject javaString = QAndroidJniObject::fromString(errMsg);
+            QAndroidJniObject toast = QAndroidJniObject::callStaticObjectMethod("android/widget/Toast", "makeText",
+                                                                                "(Landroid/content/Context;Ljava/lang/CharSequence;I)Landroid/widget/Toast;",
+                                                                                QtAndroid::androidActivity().object(),
+                                                                                javaString.object(),
+                                                                                jint(1));
+            toast.callMethod<void>("show");
+        });
+#endif
```

Comment:
This should go to a separate utility class e.g. AndroidUtils, as a method showToast(QString text)

Here we should only set our internal error (mLastError) - and somewhere else whoever listens to networkErrorOccurred() should check whether last error is empty, and show the toast (on Android) / message box (otherwise)... right now on desktop build one wouldn't get any warning at all